python-test-samples/step-functions-local-mock/README.md

```diff
@@ -0,0 +1,296 @@
+```
+
+#### Run the Unit Tests
+
+``` shell
+step-functions-local-mock/tests$
+python3 -m pytest -s unit/src/test_step_functions_local.py
+```
+
+Expected output
+
+``` shell
+step-functions-local-mock/tests$
+python3 -m pytest -s unit/src/test_step_functions_local.py
+============================================================== test session starts ==============================================================
+platform linux -- Python 3.10.12, pytest-8.3.5, pluggy-1.5.0 -- /home/ubuntu/environment/step-functions-local-mock/tests/venv/bin/python3
+cachedir: .pytest_cache
+rootdir: /home/ubuntu/environment/step-functions-local-mock/tests
+collected 3 items                                                                                                                               
+
+unit/src/test_step_functions_local.py::test_happy_path Pulling image testcontainers/ryuk:0.8.1
+Container started: 86676d42c5de
+Waiting for container <Container: 86676d42c5de> with image testcontainers/ryuk:0.8.1 to be ready ...
+Pulling image amazon/aws-stepfunctions-local
+Container started: 034b00a0307a
+Waiting for container <Container: bfd3d626a311> with image amazon/aws-stepfunctions-local to be ready ...
+PASSED
+unit/src/test_step_functions_local.py::test_retry_path PASSED
+unit/src/test_step_functions_local.py::test_hybrid_path PASSED
+
+======================================================================== 3 passed in 25.94s =========================================================================
+```
+
+
+#### Clean up section
+
+> clean pyenv environment
+
+```sh
+step-functions-local-helloworld/tests$
+deactivate
+rm -rf venv/
+```
+
+> unsetting variables
+
+```sh
+unset AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
+unset AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+unset AWS_REGION='us-east-1'
+```
+
+> cleanning docker
+
+```sh
+docker ps --filter "ancestor=amazon/aws-stepfunctions-local" -q | xargs -r docker kill
+docker rmi amazon/aws-stepfunctions-local
+```
+
+#### Debug - PyTest Debugging
+
+For more detailed debugging in pytest:
+
+```sh
+# Run with verbose output and show print statements
+python3 -m pytest -s -v unit/src/test_step_functions_local.py
+
+# Run with debug logging
+python3 -m pytest -s -v unit/src/test_step_functions_local.py --log-cli-level=DEBUG
+
+# Run only a specific test
+python3 -m pytest -s -v unit/src/test_step_functions_local.py::test_hybrid_path
+python3 -m pytest -s -v unit/src/test_step_functions_local.py::test_retry_path
+```
+
+---
+
+### Fast local development for Step Functions
+
+#### AWS CLI Commands for Manual Verification
+
+If you need to manually verify the state machine or execution details, you can use these commands:
+
+#### Configure environment variables:
+
+```sh
+step-functions-local-lambda$
+export AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
+export AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+export AWS_REGION='us-east-1'
+```
+
+#### Debug state machine - 1. Start the Step Functions Local container:
+
```

Comment:
Add a note that this sample starts *above* the `/tests` directory? 

The other two are inside it, the change threw me off.